src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -199,11 +201,20 @@ private void SetAccessWriteTimes(string path, long? accessTime, long? writeTime)
             // force a refresh so that we have an up-to-date times for values not being overwritten
             _fileStatusInitialized = -1;
             EnsureStatInitialized(path);
-            Interop.Sys.UTimBuf buf;
-            // we use utime() not utimensat() so we drop the subsecond part
-            buf.AcTime = accessTime ?? _fileStatus.ATime;
-            buf.ModTime = writeTime ?? _fileStatus.MTime;
-            Interop.CheckIo(Interop.Sys.UTime(path, ref buf), path, InitiallyDirectory);
+
+            // we use utimes()/utimensat() to set the accessTime and writeTime
+            Span<Interop.Sys.TimeSpec> buf = stackalloc Interop.Sys.TimeSpec[2];
+
+            // setting second part
+            buf[0].TvSec = accessTime ?? _fileStatus.ATime;
+            buf[1].TvSec = writeTime ?? _fileStatus.MTime;
+
+            // setting nanosecond part
+            buf[0].TvNsec = accessTime == null ? _fileStatus.ATimeNsec : 0;
+            buf[1].TvNsec = writeTime == null ? _fileStatus.MTimeNsec : 0;
+
+            Interop.CheckIo(Interop.Sys.UTimensat(path, ref MemoryMarshal.GetReference(buf)), path, InitiallyDirectory);          
```

Comment:
I don't think the use of Span really buys much here.  Seems like it's attempt to avoid using unsafe code, but the UTimensat marshaling is effectively already unsafe, and the use of ref here is going to incur higher marshaling costs.  I think it'd be better to just stackalloc into a pointer, and have the interop signature continue to take a pointer rather than a ref.